toolchain/lower/file_context.cpp

```diff
@@ -263,13 +249,51 @@ auto FileContext::BuildFunctionDecl(SemIR::FunctionId function_id,
         break;
     }
   }
+}
+
+auto FileContext::BuildFunctionDecl(SemIR::FunctionId function_id,
+                                    SemIR::SpecificId specific_id)
+    -> llvm::Function* {
+  const auto& function = sem_ir().functions().Get(function_id);
+
+  // Don't lower generic functions. Note that associated functions in interfaces
+  // have `Self` in scope, so are implicitly generic functions.
+  if (function.generic_id.has_value() && !specific_id.has_value()) {
+    return nullptr;
+  }
+
+  // Don't lower builtins.
+  if (function.builtin_function_kind != SemIR::BuiltinFunctionKind::None) {
+    return nullptr;
+  }
+
+  auto implicit_param_patterns =
+      sem_ir().inst_blocks().GetOrEmpty(function.implicit_param_patterns_id);
+  // TODO: Include parameters corresponding to positional parameters.
+  auto param_patterns =
+      sem_ir().inst_blocks().GetOrEmpty(function.param_patterns_id);
+
+  // TODO: Consider tracking whether the function has been used, and only
+  // lowering it if it's needed.
+
+  const auto return_info =
+      SemIR::ReturnTypeInfo::ForFunction(sem_ir(), function, specific_id);
```

Comment:
The old code would CHECK that the return_info was valid, with `"Should not lower invalid functions."`. Could you explain why we want to handle this case now? Does an incomplete type result in an invalid `return_info`?